Orange/widgets/data/oweditdomain.py

```diff
@@ -56,27 +56,39 @@ def variable_description(var, skip_attributes=False):
                 (("values", tuple(var.values)),),
                 attributes)
     else:
-        return (var_type.__module__,
-                var_type.__name__,
-                var.name,
-                (),
-                attributes)
+        if var.is_time:
+            return (var_type.__module__,
+                    var_type.__name__,
+                    var.name,
+                    (("have_date", var.have_date),
+                     ("have_time", var.have_time)),
+                    attributes)
+        else:
+            return (var_type.__module__,
+                    var_type.__name__,
+                    var.name,
+                    (),
+                    attributes)
 
 
 def variable_from_description(description, compute_value=None):
     """Construct a variable from its description (see
     :func:`variable_description`).
 
     """
-    module, type_name, name, kwargs, attrs = description
+    module, type_name, name, args, attrs = description
     try:
         constructor = get_qualified(module, type_name)
     except (ImportError, AttributeError):
         raise ValueError("Invalid descriptor type '{}.{}"
                          "".format(module, type_name))
 
-    var = constructor(name, compute_value=compute_value, **dict(list(kwargs)))
+    kwargs = dict(list(args))
+    var = constructor(name, compute_value=compute_value, **kwargs)
     var.attributes.update(attrs)
+    if var.is_time:
+        var.have_date = kwargs['have_date']
```

Comment:
Have you considered adding have_date and have_time to TimeVariable constructor as keyword arguments?

Reply:
I was considering this but was afreide of breking something else, will add have_time and have_date as keyword arguments to Tiemariable constructor.